src/js/components/Table.jsx

```diff
@@ -1,71 +1,111 @@
+// @flow
+
 import React, { Component, PropTypes } from 'react';
 
-const { array, node, string } = PropTypes;
+function getKey(column:string | Object) {
+    if (typeof column === 'string') {
+        return column;
+    }
+    return column.label;
+}
+
+function getLabel(column:string | Object) {
+    if (typeof column === 'string') {
+        return column;
+    }
+    return column.label;
+}
+
+function getClass(column:string | Object) {
+    if (typeof column === 'string') {
+        return null;
+    }
+    return column.className;
+}
 
 /**
  * Renders a simple HTML table with optional header elements.
  *
  * Properties:
+ *
  * "children": one or more TR elements
+ *
  * "headers": an array of Strings to render, or
  *            an array of Objects with shape: { label:String, className:String }
  *
+ * "disableHeaderDivider": Optional, truthy if you wish to disable the HR row at the bottom of THEAD
+ *
+ * "disableDefaultPadding": Optional, truthy if you wish to disable default padding added by removing u-table-padding
+ * className from the TABLE element
+ *
  * To set explicit column widths, specify className for the header elements and
  * specify specify className="fixed" on the Table component to use table-layout: fixed.
+ *
+ * Specify additional className="u-highlight-rows" for whole-row mouseover hover
+ *
+ * Head / body divider will only be included automagically when using headers prop.
  */
 export class Table extends Component {
 
-    getKey(column) {
-        if (typeof column === 'string') {
-            return column;
-        }
-        return column.label;
-    }
+    render() {
+        const {
+            headers,
+            children,
+            className,
+            disableHeaderDivider,
+            disableDefaultPadding } = this.props;
 
-    getLabel(column) {
-        if (typeof column === 'string') {
-            return column;
-        }
-        return column.label;
-    }
+        const divider = headers && headers.length && !disableHeaderDivider ?
+            <TableDivider numCols={headers.length}/> : '';
```

Comment:
We should return `null` rather than empty string to avoid this warning:

`Warning: validateDOMNesting(...): Whitespace text nodes cannot appear as a child of <thead>. Make sure you don't have any extra whitespace between tags on each line of your source code. See Table > thead > #text.`